cpp/test/sg/rf_treelite_test.cu

```diff
@@ -274,23 +277,22 @@ class RfTreeliteTestClf : public RfTreeliteTestCommon<T, L> {
     // #class for multi-class classification
     this->task_category = 2;
 
-    float *weight, *noise, *temp_label_d;
+    float *weight, *temp_label_d;
     std::vector<float> temp_label_h;
 
     allocate(weight, this->params.n_cols);
-    allocate(noise, this->params.n_rows);
     allocate(temp_label_d, this->params.n_rows);
 
     Random::Rng r(1234ULL);
 
+    // Generate weight for each feature.
     r.uniform(weight, this->params.n_cols, T(0.0), T(1.0), this->stream);
-    r.uniform(noise, this->params.n_rows, T(0.0), T(10.0), this->stream);
+    // Generate noise.
+    r.uniform(temp_label_d, this->params.n_rows, T(0.0), T(10.0), this->stream);
 
-    LinAlg::gemm<float, float, float, cutlass::Shape<8, 128, 128>>(
-      CUBLAS_OP_N, CUBLAS_OP_N, this->params.n_rows, this->params.n_cols, 1,
-      1.f, this->data_d, this->params.n_rows, weight, this->params.n_cols, 1.f,
-      noise, this->params.n_rows, temp_label_d, this->stream);
-    CUDA_CHECK(cudaStreamSynchronize(this->stream));
+    LinAlg::gemv<float>(this->data_d, this->params.n_cols, this->params.n_rows,
+                        weight, temp_label_d, false, 1.f, 1.f,
```

Comment:
Why is transpose false here, but true for regression?

Reply:
Transpose should be false for all cases. However, the `lda` needs be set to `n_cols`. In the gemv implementation, the `lda` is always be set to `n` which is `n_rows` in this case (https://github.com/rapidsai/cuml/blob/bbb8e6ceac284bfaeb7b2a902ec73135cedcf026/cpp/src_prims/linalg/gemv.h#L47). We can apply a transpose before calling the `gemv` to solve this or write a custom `gemv` which don't set `lda` based on transpose or not (let the user decide what's the value of `lda`). Which one do you prefer?